### PR TITLE
install python3 on jenkins master and workers

### PR DESCRIPTION
**Pull request by estute**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

-----------

This installs python3.5 (https://packages.ubuntu.com/xenial/python/python3.5) on build jenkins master and jenkins workers. It does not set it as the default python, but makes it present for the ShiningPanda plugin to create virtual environments

### CODE REVIEW

**Comment by estute** [comment]
Building here: https://build.testeng.edx.org/job/build-packer-ami/10895/

File: playbooks/roles/jenkins_build/defaults/main.yml [hunk 1]
```diff
@@ -2,6 +2,10 @@ build_jenkins_user_uid: 1002
 build_jenkins_group_gid: 1004
 build_jenkins_version: jenkins_2.89.4
 build_jenkins_jvm_args: '-Djava.awt.headless=true -Xmx16384m -DsessionTimeout=60'
+
+build_jenkins_python_versions:
+  - python3.5
```
Comment (jmbowman) [inline playbooks/roles/jenkins_build/defaults/main.yml]: Is this sufficient, or do we need `python3.5-dev` for installation of packages with C extensions?

File: playbooks/roles/jenkins_worker/tasks/python.yml [hunk 2]
```diff
@@ -3,3 +3,11 @@
 # Requests library is required for the github status script.
 - name: Install requests Python library
   pip: name=requests state=present
+
+# Install Python 3.5. For now, do not set it as the default python
+# version
+- name: Install python 3.5
+  apt:
+    name: 'python3.5'
```
Comment (jmbowman) [inline playbooks/roles/jenkins_worker/tasks/python.yml]: There's a `jenkins_debian_pkgs` list in `jenkins_worker/defaults/main.yml`; should this (or the `-dev` variant) just be added there instead?

Comment (estute) [reply playbooks/roles/jenkins_worker/tasks/python.yml]: I thought about that, and but kept it separate just to draw attention to it, as it is perhaps the most important application/tool we use

File: playbooks/roles/jenkins_build/defaults/main.yml (hunk 1 above)
Comment (estute) [reply playbooks/roles/jenkins_build/defaults/main.yml]: That's a fair point. Fixing.

**Review by jmbowman — APPROVED** [review]

**Pull request merged**